libraries/botbuilder-dialogs-adaptive/src/conditions/onChooseIntent.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Dialog, TurnPath } from 'botbuilder-dialogs';
+import { Expression, ExpressionParserInterface } from 'botframework-expressions';
+import { OnIntent, OnIntentConfiguration } from './onIntent';
+
+export interface OnChooseIntentConfiguration extends OnIntentConfiguration {
+    intents?: string[];
+}
+
+export class OnChooseIntent extends OnIntent {
+
+    public static declarativeType = 'Microsoft.OnChooseIntent';
+
+    public intents: string[] = [];
+
+    public constructor(actons: Dialog[] = [], condition?: string) {
+        super('chooseintent', [], actons, condition);
+    }
+
+    public configure(config: OnChooseIntentConfiguration): this {
+        return super.configure(config);
+    }
+
+    public getExpression(parser: ExpressionParserInterface): Expression {
+        if (this.intents.length > 0) {
```

Comment:
can this.intents be null?

Reply:
@luhan2017 The initial value of *this.intents* is [], an empty string array, unless being overwritten, it can't be null or undefined by default.